generators/server/templates/src/main/java/package/repository/_UserRepository.java

```diff
@@ -41,9 +39,11 @@
     Optional<User> findOneByLogin(String login);
     <%_ if (databaseType == 'sql') { _%>
 
-    @Query(value = "select distinct user from User user left join fetch user.authorities",
-        countQuery = "select count(user) from User user")
-    Page<User> findAllWithAuthorities(Pageable pageable);
+    @EntityGraph(attributePaths = "authorities")
+    User findOneWithAuthoritiesById(<%= pkType %> id);
+
+    @EntityGraph(attributePaths = "authorities")
+    Optional<User> findOneWithAuthoritiesByLogin(String login);
```

Comment:
Yes you have to create one query by need, in order to fetch only what you need for your purpose. That way it's always optimized and you never grab all the database. Plus, with Spring Data JPA, you have to define every query like `findOneByCode`, `findOneByLabel`, `findOneByName`, ... to get what your want.
For the name, if you need to create a query with several relations, you can named it something like `findOneWithRelationsById` or `findOneForManagementById` to don't have a very long name.
Do you think it can be simplified?